### DISCUSSION TITLE
unique column

### DISCUSSION THREAD

**Discussion by NEBA0317** [supabase/supabase#2461] [Questions] [answered]
how to make unique column for email ?

**Comment by silentworks** [comment, marked as answer]
You can either add a UNIQUE constraint when creating the table by doing:

```sql
CREATE TABLE users (
	email VARCHAR (16) UNIQUE NOT NULL,
	user_id uuid NOT NULL
);
```

or you can alter an existing table to make a column unique:

```sql
ALTER TABLE ONLY users
	ADD CONSTRAINT unique_email UNIQUE (email);
```

**Reply by steve-chavez** [reply]
In addition to UNIQUE, I'd also advice to make the column case-insensitive, so `one@email.com = ONE@email.com`. For this you can use the [citext](https://www.postgresql.org/docs/current/citext.html) type:

```sql
CREATE TABLE users (
	email citext UNIQUE NOT NULL,
	user_id uuid NOT NULL
);
```
